app/models/metric/ci_mixin/capture.rb

```diff
@@ -76,56 +78,59 @@ def calculate_gap(interval_name, start_time)
     end
   end
 
-  def just_perf_capture(interval_name, start_time = nil, end_time = nil)
+  def just_perf_capture(interval_name, start_time, end_time, metrics_capture)
     log_header = "[#{interval_name}]"
     log_time = ''
     log_time << ", start_time: [#{start_time}]" unless start_time.nil?
     log_time << ", end_time: [#{end_time}]" unless end_time.nil?
 
     _log.info("#{log_header} Capture for #{log_target}#{log_time}...")
 
-    start_range = end_range = counters = counter_values = counters_data = nil
+    start_range = end_range = counters_by_mor = counter_values_by_mor_and_ts = target_ems = nil
+    counters_data = {}
     _, t = Benchmark.realtime_block(:total_time) do
-      Benchmark.realtime_block(:capture_state) { perf_capture_state }
+      Benchmark.realtime_block(:capture_state) { VimPerformanceState.capture_many(Array(metrics_capture.target)) }
```

Comment:
Wonder if just adding array support to the existing `VimPerformanceState.capture` method?  Are there any other callers of `VimPerformanceState.capture` that we need to maintain the old behavior for?